Source/WebCore/platform/calc/CalcOperator.h

```diff
@@ -63,4 +65,239 @@ enum class CalcOperator : uint8_t {
+        float right = evaluate(children[1]);
+        if (!right)
+            return std::numeric_limits<double>::quiet_NaN();
+        if ((left < 0) == (right < 0))
+            return std::fmod(left, right);
+        return std::remainder(left, right);
+    }
+    case CalcOperator::Rem: {
+        if (children.size() != 2)
+            return std::numeric_limits<double>::quiet_NaN();
+        float left = evaluate(children[0]);
+        float right = evaluate(children[1]);
+        if (!right)
+            return std::numeric_limits<double>::quiet_NaN();
+        return std::fmod(left, right);
+    }
+    case CalcOperator::Round:
+        return std::numeric_limits<double>::quiet_NaN();
+    case CalcOperator::Up: {
+        if (children.size() != 2)
+            return std::numeric_limits<double>::quiet_NaN();
+        auto valueToRound = evaluate(children[0]);
+        auto roundingInterval = evaluate(children[1]);
+        if (!std::isinf(valueToRound) && std::isinf(roundingInterval)) {
+            if (!valueToRound)
+                return valueToRound;
+            return std::signbit(valueToRound) ? -0.0 : std::numeric_limits<double>::infinity();
+        }
+        auto ret = getNearestMultiples(valueToRound, roundingInterval);
+        return ret.second;
+    }
+    case CalcOperator::Down: {
+        if (children.size() != 2)
+            return std::numeric_limits<double>::quiet_NaN();
+        auto valueToRound = evaluate(children[0]);
+        auto roundingInterval = evaluate(children[1]);
+        if (!std::isinf(valueToRound) && std::isinf(roundingInterval)) {
+            if (!valueToRound)
+                return valueToRound;
+            return std::signbit(valueToRound) ? -std::numeric_limits<double>::infinity() : +0.0;
+        }
+        auto ret = getNearestMultiples(valueToRound, roundingInterval);
+        return ret.first;
+    }
+    case CalcOperator::Nearest: {
+        if (children.size() != 2)
+            return std::numeric_limits<double>::quiet_NaN();
+        auto valueToRound = evaluate(children[0]);
+        auto roundingInterval = evaluate(children[1]);
+        if (!std::isinf(valueToRound) && std::isinf(roundingInterval))
+            return std::signbit(valueToRound) ? -0.0 : +0.0;
+        auto ret = getNearestMultiples(valueToRound, roundingInterval);
+        auto upperB = ret.second;
+        auto lowerB = ret.first;
+        return std::abs(upperB - valueToRound) <= std::abs(roundingInterval) / 2 ? upperB : lowerB;
+    }
+    case CalcOperator::ToZero: {
+        if (children.size() != 2)
+            return std::numeric_limits<double>::quiet_NaN();
+        auto valueToRound = evaluate(children[0]);
+        auto roundingInterval = evaluate(children[1]);
+        if (!std::isinf(valueToRound) && std::isinf(roundingInterval))
+            return std::signbit(valueToRound) ? -0.0 : +0.0;
+        auto ret = getNearestMultiples(valueToRound, roundingInterval);
+        auto upperB = ret.second;
```

Comment:
`auto [lower, upper] = nearest` would be better.

Reply:
Fixed